### ISSUE TITLE
Add solc compiler support.

### ISSUE THREAD

**Issue by UrsZeidler** [webdaford/smart-contract-tools#6]
We should add the possibility to compile the code with the solidity compiler. 
I have some basic code for this(quite trivial), it's currently based on preferences and should be based on project properties, like the java compiler.
XText provides this already, _com.dell.research.bc.eth.solidity.editor.Solidity.compiler.propertyPage_ and this would be the right place to integrate this.

In  _com.dell.research.bc.eth.solidity.ui_ we add a builder which take the project preferences to compiles the code.

**Comment by webdaford** [comment]
Agreed.  The full vision is that this would be a complete smart contract development environment with full compilation and debug support.  I believe that it is also possible to use Xtext to generate Java from Solidity (!) and then debug the execution of that with full integration at the source level with Solidity.

**Comment by UrsZeidler** [comment]
Maybe we get the https://github.com/ethereum/ethereumj people on the eclipse boat. This would gives us network access (deployment) also there got the etherum VM.
I think will file them a feature request.

**Comment by UrsZeidler** [comment]
I have opened the issue (https://github.com/ethereum/ethereumj/issues/379).
